imageregistry/v1/types.go

```diff
@@ -283,6 +291,11 @@ type ImageRegistryConfigStorage struct {
 	// azure represents configuration that uses Azure Blob Storage.
 	// +optional
 	Azure *ImageRegistryConfigStorageAzure `json:"azure,omitempty"`
+	// managementState indicates if the operator manages the underlying
+	// storage unit.
+	// +optional
+	// +kubebuilder:validation:Pattern=`^(Managed|Unmanaged)$`
+	ManagementState string `json:"managementState,omitempty"`
```

Comment:
since this is optional, what does the empty value default mean?

Reply:
Being practical there is no difference between "Unmanaged" and empty value. During operator cycles the value from "Status.StorageManaged" would by copied appropriately to it if it is empty.

Reply:
Hm, copying from status to spec is definitely unusual.  But that basically sounds like a one-time migration step (the only time this should be unset is on an upgrade, right?  otherwise i'd expect it to always be set by the bootstrap logic).

Reply:
Yes, needed during upgrades, once it is set we won't touch it anymore. On a fresh install it will be set during initial storage creation and never touched again by the operator.